### PR TITLE
Fix #2471 and #2540

### LINKED ISSUE #2471: Address search for Liechtenstein not working
can't use address search for Liechtenstein , when i type the city/zipcode in for one of the 11 town in this small country, no of them appearing on the screen.
screencast_2016-04-17-14-56-01
Reverse geocoding is working and
 when i use inspector the output seams correct.
```
/tmp/testmap$ ./inspector.sh -vaddress Liechtenstein_europe.obf |grep City
		City [VILLAGE] Balzers 240100249	(111)
		City [VILLAGE] Eschen 3022807519	(130)
		City [VILLAGE] Gamprin 240080206	(40)
		City [VILLAGE] Gamprin-Bendern 240070712	(15)
		City [VILLAGE] Mauren 3022807520	(83)
		City [VILLAGE] Nendeln 240046176	(11)
		City [VILLAGE] Planken 32011242	(19)
		City [VILLAGE] Rotenboden 935278226	(8)
		City [VILLAGE] Ruggell 240120066	(53)
		City [VILLAGE] Schaan 240079149	(152)
		City [VILLAGE] Schaanwald 240060684	(10)
		City [VILLAGE] Schellenberg 240077147	(47)
		City [VILLAGE] Triesen 240092661	(94)
		City [VILLAGE] Triesenberg 240106262	(122)
		City [VILLAGE] Vaduz 1932181216	(163)
		City [HAMLET] Ebenholz 3577169505	(5)
		City [HAMLET] Malbun 347290636	(4)
		City [HAMLET] Oberplanken 32011241	(2)
		City [HAMLET] Steg 1796218506	(2)

```
This screencast is record on osmand~ 2.3.3. but this issue was existing on earlier version to. (include google play verion of it)
Map tested with :
-self build map  with OsmAndMapCreator.sh:
-http://download.osmand.net/list.php ( version 01.03.2016)
-osmand inapp download (version 01.03.2016)

### PR DESCRIPTION
**Pull request by rominf**
Also show all villages if no cities are present in region

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/resources/RegionAddressRepositoryBinary.java [hunk 1]
```diff
@@ -229,35 +271,11 @@ public synchronized void preloadStreets(City o, ResultMatcher<Street> resultMatc
 				}
 			}
 
-			int initialsize = citiesToFill.size();
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/resources/RegionAddressRepositoryBinary.java]: spelling error

Comment (rominf) [inline OsmAnd/src/net/osmand/plus/resources/RegionAddressRepositoryBinary.java]: My bad. Change it to `initialSize`?

Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/resources/RegionAddressRepositoryBinary.java]: Yes,  it would be more logical.

Comment (rominf) [inline OsmAnd/src/net/osmand/plus/resources/RegionAddressRepositoryBinary.java]: I fixed this: https://github.com/osmandapp/Osmand/commit/ed5b746392548861f9b5bbdb4842de6d5dfaa1af

**Pull request merged**